### PR TITLE
Added new feature of viewing combat history and tests for it

### PR DESCRIPTION
**Pull request by nikolalsvk**

### CODE REVIEW

File: app/assets/javascripts/char_combat.js [hunk 1]
```diff
@@ -0,0 +1,8 @@
+$(document).on('page:load', ready);
```
Comment (nikolalsvk) [inline app/assets/javascripts/char_combat.js]: Delete this file.

File: app/controllers/char_combats_controller.rb [hunk 2]
```diff
@@ -27,4 +40,13 @@ def find_first_char
   def find_second_char
     @second_char = current_user.char_types.find_by_id(params["char_combat"]["second_char"])
   end
+
+  def who_won(winner)
```
Comment (nikolalsvk) [inline app/controllers/char_combats_controller.rb]: You can move this to the model.

File: app/views/char_combats/index.html.erb [hunk 3]
```diff
@@ -1,7 +1,22 @@
...
+    <tr>
+      <td>Fight <%= combat.id %></td>
+      <td><%= combat.winner %></td>
+      <td><%= combat.loser %></td>
+      <td><%= link_to "Show", char_combat_path(combat) %></td>
+      <td><%= link_to "Destroy", char_combat_path(combat), :method => :delete,
+              data: { confirm: 'Are you sure?' } %>
+    </tr>
+  <% end %>
+</table>
```
Comment (nikolalsvk) [inline app/views/char_combats/index.html.erb]: Add new line here.

File: features/support/env.rb [hunk 4]
```diff
@@ -53,6 +53,7 @@
 
 # Possible values are :truncation and :transaction
 Capybara.javascript_driver = :webkit
+Capybara.default_wait_time = 5
```
Comment (nikolalsvk) [inline features/support/env.rb]: Remove this line.

File: spec/factories/char_combats.rb [hunk 5]
```diff
@@ -0,0 +1,7 @@
+FactoryGirl.define do
+  factory :char_combat do
+    winner "MyString"
+loser "MyString"
```
Comment (nikolalsvk) [inline spec/factories/char_combats.rb]: Fix indentation.

File: app/models/char_combat.rb [hunk 6]
```diff
@@ -0,0 +1,21 @@
+class CharCombat < ActiveRecord::Base
+  validates :first_combatant, :presence => true
+  validates :second_combatant, :presence => true
+
+  before_create :combat
+
+  def combat
+    self.winner = first.attack_points > second.attack_points ? 
+      first.title : second.title
+    self.first_combatant = first.title
```
Comment (pyromaniackeca) [inline app/models/char_combat.rb]: I think you don't need this code, try removing it.

File: app/models/char_type.rb [hunk 7]
```diff
@@ -4,6 +4,7 @@ class CharType < ActiveRecord::Base
   mount_uploader :avatar, PictureUploader
   belongs_to :user
   has_many :char_attributes, :dependent => :destroy
+  has_many :char_combats
```
Comment (nikolalsvk) [inline app/models/char_type.rb]: This needs to be `:dependent => :destroy`

File: app/views/char_combats/_char_combats.html.erb [hunk 8]
```diff
@@ -0,0 +1,19 @@
...
+    <tr>
+      <td>Fight <%= char_combat.id %></td>
+      <td><%= render "char_combats/winner", :char_combat => char_combat %></td>
+      <td><%= render "char_combats/loser", :char_combat => char_combat %></td>
+      <td><%= link_to "Show", char_combat_path(char_combat) %></td>
+      <td><%= link_to "Destroy", char_combat_path(char_combat), :method => :delete,
+              :data => { :confirm => 'Are you sure?' } %>
+    </tr>
+  <% end %>
+</table>
```
Comment (nikolalsvk) [inline app/views/char_combats/_char_combats.html.erb]: Add new line here.

File: app/views/char_combats/_loser.html.erb [hunk 9]
```diff
@@ -0,0 +1,5 @@
+<% if char_combat.winner == char_combat.second_combatant.id %>
+  <%= char_combat.first_combatant.title %>
+<% else %>
+  <%= char_combat.second_combatant.title %>
+<% end %>
```
Comment (nikolalsvk) [inline app/views/char_combats/_loser.html.erb]: Add new line here.

File: app/views/char_combats/_winner.html.erb [hunk 10]
```diff
@@ -0,0 +1,5 @@
+<% if char_combat.winner == char_combat.first_combatant.id %>
+  <%= char_combat.first_combatant.title %>
+<% else %>
+  <%= char_combat.second_combatant.title %>
+<% end %>
```
Comment (nikolalsvk) [inline app/views/char_combats/_winner.html.erb]: Add new line here.

Comment (pyromaniackeca) [inline app/views/char_combats/_winner.html.erb]: Add new line here.

**Pull request merged**